backend/src/v2/component/launcher_v2.go

```diff
@@ -696,3 +712,16 @@ func prepareOutputFolders(executorInput *pipelinespec.ExecutorInput) error {
 
 	return nil
 }
+
+func addDefaultParams(
+	executorInput *pipelinespec.ExecutorInput,
+	component *pipelinespec.ComponentSpec,
+) *pipelinespec.ExecutorInput {
+	for name, value := range component.GetInputDefinitions().GetParameters() {
+		_, hasInput := executorInput.GetInputs().GetParameterValues()[name]
+		if value.GetDefaultValue() != nil && !hasInput {
+			executorInput.GetInputs().GetParameterValues()[name] = value.GetDefaultValue()
```

Comment:
As you update `executorInput` in place, there could be potentially no difference whether you add default values here vs. earlier at this point: https://github.com/kubeflow/pipelines/blob/c0640d7a033b6fa2133b92c2a459d9eb3dabee32/backend/src/v2/driver/driver.go#L824

And if I read the `execute` function correctly, this change reverts our earlier decision on not passing in default value explicitly. Did you see such behavior change when testing pipeline with default value?

Reply:
We had some discussion offline on this. The executor input displayed at the UI comes from the file that driver writes, so changing it in the launcher does not affect that. However, it is still not advisable to alter `executorInput`, in case it may affect later functionalities. Thus I created a deep copy of `executorInput` for this purpose. I also added a few unit tests, although they may bring some flakiness for developers outside of Google, due to authentication issue caused by opening bucket.